src/control/BarrettHandKinematicSimulationPositionCommandExecutor.cpp

```diff
@@ -27,50 +30,61 @@ BarrettHandPositionCommandExecutor::BarrettHandPositionCommandExecutor(
   }
 
   if (!mSpreadCommandExecutor)
-    throw std::invalid_argument("SpreadCommandExecutor is null.");  
+    throw std::invalid_argument("SpreadCommandExecutor is null.");
+
+  if (!mCollideWith)
+    throw std::invalid_argument("CollideWith is null.");
+
 }
 
 //=============================================================================
-std::future<void> BarrettHandPositionCommandExecutor::execute(
-  Eigen::Matrix<double, 4, 1> _positions,
-  ::dart::collision::CollisionGroupPtr _collideWith)
+std::future<void> BarrettHandKinematicSimulationPositionCommandExecutor
+  ::execute(const Eigen::VectorXd& goalPositions)
 {
-  if (!_collideWith)
-    throw std::invalid_argument("CollideWith is null.");
-
   std::lock_guard<std::mutex> lockSpin(mMutex);
-  
+
   if (mInExecution)
     throw std::runtime_error("Another command in execution.");
 
+  if (goalPositions.size() != 4)
+  {
+    std::stringstream message;
+    message << "GoalPositions must have 4 elements, but ["
+      << goalPositions.size() << "] given.";
+    throw std::invalid_argument(message.str());
+  }
+
   mPromise.reset(new std::promise<void>());
-  mProximalGoalPositions = _positions.topRows(3); 
-  mSpreadGoalPosition = _positions(3);
-  mCollideWith = _collideWith;
+  mProximalGoalPositions = goalPositions.topRows(3);
```

Comment:
Nit: `topRows()` is a method for matrices. We could the method for vectors as `head<3>()`